test/catch2/ext/test_key_provider.cpp

```diff
@@ -0,0 +1,265 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "ext/test/key_provider/key_provider.h"
+
+#include "wrappers/mock_session.h"
+#include "utils.h"
+
+#include "wiredtiger.h"
+#include "wt_internal.h"
+
+#include <catch2/catch.hpp>
+
+#include <functional>
+#include <iostream>
+#include <memory>
+
+/*
+ * kp_fixture
+ *     Test fixture for the key provider extension tests.
+ */
+struct kp_fixture {
+    utils::shared_library lib{KEY_PROVIDER_EXTENSION};
+
+    using extension_init_t = decltype(&wiredtiger_extension_init);
+    extension_init_t extension_init = nullptr;
+
+    static constexpr char conn_config[] = "create,in_memory";
+    WT_CONNECTION *conn = nullptr;
+
+    KEY_PROVIDER *kp = nullptr;
+    using kp_ptr_t = std::unique_ptr<KEY_PROVIDER, std::function<void(KEY_PROVIDER *)>>;
+
+    ~kp_fixture()
+    {
+        reset();
+
+        if (conn != nullptr) {
+            conn->close(conn, nullptr);
+            conn = nullptr;
+        }
+    }
+
+    kp_fixture() : extension_init(lib.get<extension_init_t>("wiredtiger_extension_init"))
+    {
+        int ret = wiredtiger_open(DB_HOME, nullptr, conn_config, &conn);
```

Comment:
I don't think we need a real connection object here. Could we re-use our mock connection here?